node_exporter.go

```diff
@@ -89,7 +89,7 @@ func main() {
 	}
 
 	// TODO(ts): Remove deprecated and problematic InstrumentHandlerFunc usage.
-	http.HandleFunc(*metricsPath, prometheus.InstrumentHandlerFunc("prometheus", handler))
+	http.Handle(*metricsPath, promhttp.Handler())
```

Comment:
This is for standard exposition, which is not what the node exporter is doing. We still need the custom handler function we have.

Reply:
I don't think this is necessary, I register the metrics in the collector setup above.  I tested with end-to-end, everything looks correct.

Reply:
Oh, I see, without the handler, the `collect[]` param stops working.